lib/ui/ui_dart_state.cc

```diff
@@ -79,10 +80,20 @@ const TaskRunners& UIDartState::GetTaskRunners() const {
 }
 
 fml::RefPtr<flutter::SkiaUnrefQueue> UIDartState::GetSkiaUnrefQueue() const {
-  if (!io_manager_) {
-    return nullptr;
-  }
-  return io_manager_->GetSkiaUnrefQueue();
+  fml::RefPtr<flutter::SkiaUnrefQueue> unref_queue;
```

Comment:
I think it might be better to just assert that the caller is doing this on the right thread rather than posting a task for them.

IOW, let's just do an `FML_DCHECK(IOTaskRunner->RunsTasksOnCurrentThread())` and fix any sites that aren't doing what they should.

Reply:
makes sense. I'll add the same assertion to the GetResourceContext() method too, and add a comment in the header saying these two methods need to be called off the IO thread.

Reply:
I pushed the IO taskrunner one level higher up into the `CreateGPUObject` static method. I'm not convinced it's reasonable to expect callers of that method to do so on the IO thread.